observability/logging/logging-6.2/log6x-configuring-lokistack-otlp-6.2.adoc

```diff
@@ -0,0 +1,145 @@
+
+When using the {loki-op} in `openshift-logging` mode, attribute mapping follow OpenShift defaults, but custom mappings can be configured to adjust these. Custom mappings allow further configurations to meet specific needs.
+
+In `openshift-logging` mode, custom attribute mappings can be configured globally for all tenants or for individual tenants as needed. When custom mappings are defined, they are appended to the OpenShift defaults. If default recommended labels are not required, they can be disabled in the tenant configuration.
+
+[NOTE]
+====
+A major difference between the {loki-op} and Loki itself lies in inheritance handling. Loki only copies `default_resource_attributes_as_index_labels` to tenants by default, while the {loki-op} applies the entire global configuration to each tenant in `openshift-logging` mode.
+====
+
+Within `LokiStack`, attribute mapping configuration is managed through the `limits` setting:
+
+[source,yaml]
+----
+# ...
+spec:
+  limits:
+    global:
+      otlp: {} # <1>
+    tenants:
+      application:
+        otlp: {} # <2>
+----
+<1> Global OTLP attribute configuration.
+<2> OTLP attribute configuration for the `application` tenant within `openshift-logging` mode.
+
+[NOTE]
+====
+Both global and per-tenant OTLP configurations can map attributes to stream labels. At least one stream label is required to save a log entry to Loki storage, so ensure this configuration meets that requirement.
+====
+
+Stream labels derive only from resource-level attributes, which the `LokiStack` resource structure reflects:
+
+[source,yaml]
+----
+spec:
+  limits:
+    global:
+      otlp:
+        streamLabels:
+          resourceAttributes:
+          - name: "k8s.namespace.name"
+          - name: "k8s.pod.name"
+          - name: "k8s.container.name"
+----
+
+You can drop attributes of type resource, scope or log from the log entry.
+
+[source,yaml]
+----
+# ...
+spec:
+  limits:
+    global:
+      otlp:
+        streamLabels:
+          # ...
+        drop:
+          resourceAttributes:
+          - name: "process.command_line"
+          - name: "k8s\\.pod\\.labels\\..+"
+            regex: true
+          scopeAttributes:
+          - name: "service.name"
+          logAttributes:
+          - name: "http.route"
+----
+
+[TIP]
+====
+Use regular expressions by setting `regex: true` to apply a configuration for attributes with similar names.
+====
+
+[IMPORTANT]
+====
+Avoid using regular expressions for stream labels, as this can increase data volume.
+====
+
+Attributes that are not explicitly set as stream labels or dropped from the entry are saved as structured metadata by default.
+
+[id="customizing-openshift-defaults_{context}"]
+=== Customizing OpenShift defaults
+
+In `openshift-logging` mode, certain attributes are required and cannot be removed from the configuration due to their role in OpenShift functions. Other attributes, labeled *recommended*, might be disabled if performance is impacted. 
```

Comment:
Also a link to the data OTEL data model might be necessary here otherwise users will not get the "recommended" reference
```suggestion
In `openshift-logging` mode, certain attributes are required and cannot be removed from the configuration due to their role in OpenShift functions. Other attributes, labeled *recommended*, might be dropped if performance is impacted. 
```

Reply:
updated. And link to this? https://docs.openshift.com/container-platform/4.17/observability/logging/logging-6.1/log6x-opentelemetry-data-model-6.1.html#attributes_log6x-opentelemetry-data-model-6-1

Reply:
Yup that's it sorry if it wasn't clear

Reply:
thaks, done